x-pack/metricbeat/module/gcp/metadata_cache.go

```diff
@@ -0,0 +1,97 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package gcp
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"cloud.google.com/go/compute/apiv1/computepb"
+	"cloud.google.com/go/redis/apiv1/redispb"
+	"google.golang.org/api/dataproc/v1"
+	"google.golang.org/api/sqladmin/v1"
+
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+// Cache represents a generic cache with metadata and mutex
+type Cache[T any] struct {
+	data            map[string]T
+	lastRefreshed   time.Time
+	refreshInterval time.Duration
+	lock            sync.Mutex
+	logger          *logp.Logger
+}
+
+func (c *Cache[T]) isExpired() bool {
+	// A zero or negative interval means the cache is always considered expired.
+	if c.refreshInterval <= 0 {
```

Comment:
this is fine, but the check below also covers this case since the result of `time.Since` will always be positive.

Reply:
without this check the Windows packaging tests always failed, it's some clock issue specific to windows

Reply:
weird. but ok, it's not hurting anything.